managed-signers/src/utils/generateSignature.ts

```diff
@@ -0,0 +1,51 @@
+import { mnemonicToAccount } from "viem/accounts";
+import { getFid } from "./getFid";
+
+export const generate_signature = async function (public_key: string) {
+  if (typeof process.env.FARCASTER_DEVELOPER_MNEMONIC === "undefined") {
+    throw new Error("FARCASTER_DEVELOPER_MNEMONIC is not defined");
+  }
+
+  const FARCASTER_DEVELOPER_MNEMONIC = process.env.FARCASTER_DEVELOPER_MNEMONIC;
+  const FID = await getFid();
+
+  // DO NOT CHANGE ANY VALUES IN THIS CONSTANT
+  const SIGNED_KEY_REQUEST_VALIDATOR_EIP_712_DOMAIN = {
+    name: "Farcaster SignedKeyRequestValidator",
+    version: "1",
+    chainId: 10,
+    verifyingContract:
+      "0x00000000fc700472606ed4fa22623acf62c60553" as `0x${string}`,
+  };
+
+  // DO NOT CHANGE ANY VALUES IN THIS CONSTANT
+  const SIGNED_KEY_REQUEST_TYPE = [
+    { name: "requestFid", type: "uint256" },
+    { name: "key", type: "bytes" },
+    { name: "deadline", type: "uint256" },
+  ];
+
+  const account = mnemonicToAccount(FARCASTER_DEVELOPER_MNEMONIC);
+
+  // Generates an expiration date for the signature
+  // e.g. 1693927665
+  const deadline = Math.floor(Date.now() / 1000) + 86400; // signature is valid for 1 day from now
+  // You should pass the same value generated here into the POST /signer/signed-key Neynar API
+
+  // Generates the signature
+  const signature = await account.signTypedData({
```

Comment:
there's a function in @farcaster/core that simplifies this code dramatically. let's switch to using that